tests/chainer_tests/functions_tests/array_tests/test_as_strided.py

```diff
@@ -0,0 +1,354 @@
+import unittest
+
+from chainer import testing, Variable, grad
+
+import numpy as np
+
+from chainer.functions import as_strided
+from chainer.functions.array.as_strided import _stride_array
+
+
+@testing.parameterize(
+    {'dtype': np.float16},
+    {'dtype': np.float32},
+    {'dtype': np.float64},
+    {'dtype': np.int16},
+    {'dtype': np.int32},
+    {'dtype': np.int64}
+)
+class TestStrideArray(unittest.TestCase):
+    def test_flip_cpu(self):
+        x = np.arange(4, dtype=self.dtype)
+        y = _stride_array(x, (4,), (-1,), 3)  # [3, 2, 1, 0]
+        y_expected = np.flip(x, axis=0)
+        assert (y == y_expected).all()
+
+    @testing.attr.gpu
+    def test_flip_gpu(self):
+        import cupy as cp
+        x = cp.arange(4, dtype=self.dtype)
+        y = _stride_array(x, (4,), (-1,), 3)
+        y_expected = cp.flip(x, axis=0)
+        assert (y == y_expected).all()
```

Comment:
```suggestion
        testing.assert_allclose(y, y_expect)
```

Reply:
@okuta I considered to use `testing.assert_allclose`, but this is just a striding (i.e. pointing the exactly same memory), therefore I think every elements should be exactly the same.
I want to hear your opinion.

Reply:
@okuta fixed